tests/python/unittest/test_tir_ptx_mma.py

```diff
@@ -0,0 +1,1356 @@
+        )
+
+
+@tvm.testing.requires_cuda
+def test_gemm_mma_m8n8k32_row_col_s4u4s32():
+    sch = tvm.tir.Schedule(gemm_mma_m8n8k32_row_col_s4u4s32)
+    arch = tvm.contrib.nvcc.get_target_compute_version()
+    major, minor = tvm.contrib.nvcc.parse_compute_version(arch)
+    if major * 10 + minor < 75:
+        # Require at least SM75
+        return
+    cuda_mod = tvm.build(sch.mod, target="cuda")
+
+    ctx = tvm.cuda()
+    A_tvm = tvm.nd.empty([8, 32], "int4", ctx)
+    B_tvm = tvm.nd.empty([8, 32], "uint4", ctx)
+    C_tvm = tvm.nd.empty([8, 8], "int32", ctx)
+
+    cuda_mod(A_tvm, B_tvm, C_tvm)
+    # Currently the correctness is not checked.
+    # TODO: add correctness checking here.
+
+
+@T.prim_func
+def gemm_mma_m16n8k8_row_col_fp16fp16fp32(a: T.handle, b: T.handle, c: T.handle):
+    T.func_attr({"global_symbol": "default_function", "tir.noalias": True})
+    A = T.match_buffer(a, [16, 8], dtype="float16")
+    B = T.match_buffer(b, [8, 8], dtype="float16")
+    C = T.match_buffer(c, [16, 8], dtype="float32")
+    brow = T.env_thread("blockIdx.y")
+    bcol = T.env_thread("blockIdx.x")
+    tx = T.env_thread("threadIdx.x")
+    T.launch_thread(brow, 1)
+    T.launch_thread(bcol, 1)
+    T.launch_thread(tx, 32)
+    MultiA = T.allocate([4], "float16", scope="local")
+    MultiB = T.allocate([2], "float16", scope="local")
+    Accum = T.allocate([4], "float32", scope="local")
+    for i in range(4):
+        Accum[i] = T.float32(0)
+
+    for mma_multi_a_col in T.vectorized(4):
+        MultiA[mma_multi_a_col] = A[
+            (tx % 32) // 4 + mma_multi_a_col // 2 * 8, (tx % 32) % 4 * 2 + mma_multi_a_col % 2
+        ]
+    for mma_multi_b_col in T.vectorized(4):
+        MultiB[mma_multi_b_col] = B[
+            (tx % 32) // 4 + mma_multi_b_col // 2 * 8, (tx % 32) % 4 * 2 + mma_multi_b_col % 2
+        ]
+    T.evaluate(
+        T.ptx_mma(
+            "m16n8k8",
+            "row",
+            "col",
+            "fp16",
+            "fp16",
+            "fp32",
+            MultiA,
+            0,
+            MultiB,
+            0,
+            Accum,
+            0,
+            False,
+            dtype="float32",
+        )
+    )
+    for mma_accum_c_id in range(4):
+        C[
+            (tx % 32) // 4 + mma_accum_c_id // 2 * 8, (tx % 32) % 4 * 2 + mma_accum_c_id % 2
+        ] = T.load("float32", Accum, mma_accum_c_id)
+
+
+@tvm.testing.requires_cuda
+def test_gemm_mma_m16n8k8_row_col_fp16fp16fp32():
+    sch = tvm.tir.Schedule(gemm_mma_m16n8k8_row_col_fp16fp16fp32)
+    arch = tvm.contrib.nvcc.get_target_compute_version()
+    major, minor = tvm.contrib.nvcc.parse_compute_version(arch)
+    if major < 8:
+        # Require at least SM80
+        return
+    cuda_mod = tvm.build(sch.mod, target="cuda")
+
+    A_np = np.random.uniform(-1, 1, [16, 8]).astype("float16")
+    B_np = np.random.uniform(-1, 1, [8, 8]).astype("float16")
+    C_np = np.random.uniform(-1, 1, [16, 8]).astype("float32")
```

Comment:
Should't the value of `C_np` be zeros?

Reply:
It's OK to set `C_np` to random values, although the most standard way is to set `C_np` to zeros. The results are not affected by the initial value of `C_np` because the accumulators are always initialized to zeros.

Reply:
Yeah I am worried about the implication may confuse people.

Reply:
note that random initialization does follow the convention we have in the tvm repo, so i dont think it confuses anybody. changing to zeros should good too, so i dont have strong opinion